src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs

```diff
@@ -902,6 +902,36 @@ private void ProcessDependencies(PackageBuilder builder)
             }
         }
 
+        private bool FindDependency(IPackage projectPackage, IEnumerable<Tuple<IPackage, NuGet.PackageDependency>> packagesAndDependencies)
+        {
+            // returns true if the dependency should be added to the package
+            // This happens if the dependency is not a dependency of a dependecy
+            // Or if the project dependency version is > the dependency's dependency version
```

Comment:
Is this logic a bit too simplistic? Isn't what we really care about what version would actually get installed based on the requested version? For example, if a dependency A requests dependency B of version >=1.0.11, but there is no 1.0.11 we might get 1.0.12. In that scenario, if version 1.0.12 is what we currently have, then we don't need to include that dependency in the minimal set (even though it > the specified version).